packages/core/src/CardanoNode/types/CardanoNode.ts

```diff
@@ -48,3 +48,38 @@ export interface CardanoNode {
    */
   healthCheck(): Promise<HealthCheckResponse>;
 }
+
+// Similar to Ogmios.Point, but using Cardano.BlockId opaque string for hash
+export type Point = Pick<Tip, 'hash' | 'slot'>;
+export type Origin = 'origin';
+export type TipOrOrigin = Tip | Origin;
+export type PointOrOrigin = Point | Origin;
+export type Intersection = {
+  point: PointOrOrigin;
+  tip: TipOrOrigin;
+};
+
+export enum ChainSyncEventType {
+  RollForward,
+  RollBackward
+}
+
+export type RequestNext = () => void;
+
+export interface WithRequestNext {
+  requestNext: RequestNext;
+}
+
+export interface ChainSyncRollForward extends WithRequestNext {
+  tip: Tip;
+  eventType: ChainSyncEventType.RollForward;
+  block: Block;
+}
+
+export interface ChainSyncRollBackward extends WithRequestNext {
+  eventType: ChainSyncEventType.RollBackward;
+  point: PointOrOrigin;
+  tip: TipOrOrigin;
+}
+
+export type ChainSyncEvent = ChainSyncRollForward | ChainSyncRollBackward;
```

Comment:
Consider hoisting chain sync types to `projection` too

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1486/commits/a7d9646e8d50d0c7487588371eb2e007e8395c12